### PR TITLE
chore: Deprecate analytics & transcript methods in favor of API endpoints

### PR DESCRIPTION
**Pull request by KevLehman**
OC-990
OC-992
OC-993

### CODE REVIEW

File: apps/meteor/app/livechat/server/api/v1/transcript.ts [hunk 1]
```diff
@@ -18,3 +19,26 @@ API.v1.addRoute(
 		},
 	},
 );
+
+API.v1.addRoute(
+	'livechat/transcript/:rid',
```
Comment (murtaza98) [inline apps/meteor/app/livechat/server/api/v1/transcript.ts:24]: Any chance we can have API test for this endpoint here... please 😇

File: apps/meteor/app/livechat/server/api/v1/statistics.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	{
+		authRequired: true,
+		permissionsRequired: ['view-livechat-manager'],
+		validateParams: isLivechatAnalyticsAgentOverviewProps,
+	},
+	{
+		async get() {
+			const { name, departmentId, from, to } = this.queryParams;
+
+			if (!name) {
```
Comment (murtaza98) [inline apps/meteor/app/livechat/server/api/v1/statistics.ts:19]: Wouldn't AJV already validate this?

File: apps/meteor/app/livechat/server/api/v1/statistics.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	{
+		authRequired: true,
+		permissionsRequired: ['view-livechat-manager'],
+		validateParams: isLivechatAnalyticsOverviewProps,
+	},
+	{
+		async get() {
+			const { name, departmentId, from, to } = this.queryParams;
+
+			if (!name) {
```
Comment (murtaza98) [inline apps/meteor/app/livechat/server/api/v1/statistics.ts:47]: Same here. This is already being validated by AJV

File: apps/meteor/app/livechat/server/api/v1/transcript.ts (hunk 1 above)
Comment (KevLehman) [reply apps/meteor/app/livechat/server/api/v1/transcript.ts:24]: I forgot i added this one

File: apps/meteor/tests/end-to-end/api/livechat/04-dashboards.ts [hunk 4]
```diff
@@ -6,7 +6,7 @@ import type { Response } from 'supertest';
 import { getCredentials, api, request, credentials } from '../../../data/api-data';
 import { updatePermission, updateSetting } from '../../../data/permissions.helper';
 
-describe('LIVECHAT - dashboards', function () {
+describe.only('LIVECHAT - dashboards', function () {
```
Comment (KevLehman) [inline apps/meteor/tests/end-to-end/api/livechat/04-dashboards.ts:9]:
```suggestion
describe('LIVECHAT - dashboards', function () {
```

**Review by murtaza98 — DISMISSED** [review]

File: apps/meteor/app/livechat/server/methods/discardTranscript.ts [hunk 5]
```diff
@@ -15,6 +16,9 @@ declare module '@rocket.chat/ui-contexts' {
 Meteor.methods<ServerMethods>({
 	async 'livechat:discardTranscript'(rid: string) {
 		check(rid, String);
+		methodDeprecationLogger.warn(
+			'The method "livechat:discardTranscript" is deprecated and will be removed after version v7.0.0. Use "livechat/transcript/:rid" instead.',
```
Comment (MarcosSpessatto) [inline apps/meteor/app/livechat/server/methods/discardTranscript.ts:20]:
```suggestion
			'The method "livechat:discardTranscript" is deprecated and will be removed after version v7.0.0. Use "livechat/transcript/:rid" (DELETE) instead.',
```

File: apps/meteor/tests/end-to-end/api/livechat/00-rooms.ts [hunk 6]
```diff
@@ -2009,4 +2009,58 @@ describe('LIVECHAT - rooms', function () {
...
+					message: JSON.stringify({
+						method: 'livechat:requestTranscript',
+						params: [_id, 'test@test.com', 'Transcript of your omnichannel conversation'],
+						id: 'id',
+						msg: 'method',
+					}),
+				})
+				.expect(200);
+
+			// Then, delete transcript with livechat:deleteTranscript method
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/end-to-end/api/livechat/00-rooms.ts:2059]:
```suggestion
			// Then, delete the transcript
```

**Review by MarcosSpessatto — DISMISSED** [review]
@KevLehman Let's rename the PR please, since we are creating the endpoint to discard the transcript in this PR as well.

**Review by murtaza98 — APPROVED** [review]

**Review by aleksandernsilva — APPROVED** [review]

**Pull request merged**